src/components/TodoApp/TodoApp.tsx

```diff
@@ -0,0 +1,156 @@
+
+  const areAllTodosCompleted = todos.every(({ completed }) => completed);
+  const visibleTodos = useMemo(
+    () => getFilteredTodos(filterKey, todos),
+    [todos, filterKey],
+  );
+  const activeTodos = todos.filter(({ completed }) => !completed);
+  const hasCompletedTodo = todos.some(({ completed }) => completed);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => onNewError('Unable to load todos'));
+  }, []);
+
+  const handleAllTodosToggle = () => {
+    const todosToUpdate = todos.filter(
+      ({ completed }) => completed === areAllTodosCompleted,
+    );
+
+    setTodoIdsWithLoader(
+      prevTodoIds => [...prevTodoIds, ...todosToUpdate.map(({ id }) => id)],
+    );
+
+    todosToUpdate.forEach(todo => {
+      updateTodo(todo.id, { completed: !todo.completed })
+        .then(() => {
+          const todosCopy = [...todos];
+          const searchedTodo = todos.find(
+            ({ id }) => todo.id === id,
+          ) as Todo;
+
+          searchedTodo.completed = !searchedTodo.completed;
+
+          setTodos(todosCopy);
+        })
+        .catch(() => onNewError('Unable to update a todo'))
+        .finally(() => setTodoIdsWithLoader(
+          prevTodoIds => prevTodoIds.filter((todoId) => todoId !== todo.id),
+        ));
+    });
+  };
+
+  const handleCompletedTodosDelete = () => {
+    const completedTodos = todos.filter(({ completed }) => completed);
+
+    setTodoIdsWithLoader(prevTodoIds => {
+      return [...prevTodoIds, ...completedTodos.map(({ id }) => id)];
+    });
+
+    completedTodos.forEach(todo => {
+      deleteTodo(todo.id)
+        .then(() => {
+          setTodos(prevTodos => prevTodos.filter(({ id }) => id !== todo.id));
+        })
+        .catch(() => onNewError('Unable to delete a todo'))
+        .finally(() => setTodoIdsWithLoader(
+          prevTodoIds => prevTodoIds.filter((id) => todo.id !== id),
+        ));
+    });
+  };
+
+  return (
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+          {!!todos.length && (
+            <button
+              type="button"
+              data-cy="ToggleAllButton"
+              className={classNames('todoapp__toggle-all', {
+                active: areAllTodosCompleted,
+              })}
+              aria-label="toggle_all_todos"
+              onClick={handleAllTodosToggle}
+            />
+          )}
+
+          <NewTodo
+            setTempTodo={setTempTodo}
+            tempTodo={tempTodo}
+          />
+        </header>
+
+        <TodoList
+          todos={visibleTodos}
+          tempTodo={tempTodo}
+        />
+
+        {!!todos.length && (
+          <footer className="todoapp__footer" data-cy="Footer">
+            <span className="todo-count" data-cy="TodosCounter">
+              {`${activeTodos.length} items left`}
```

Comment:
If there is one item, there should not be a plural in the text.

Reply:
Firstly, I wrote a code that would not use plural to a single todo, but tests required it to be plural